### DISCUSSION TITLE
Charting and data visualization

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#430] [Ideas]
What are some good charting and data visualization products that you recommend?

**Comment by DaveSkender** [comment]
Since I'm mostly developing Angular websites lately, I like [Charts.js](https://www.chartjs.org) because it works natively in Javascript and does not require special wrappers.  Our [demo](https://stock-charts.azurewebsites.net) uses Chart.js.  I also use the [chartjs-chart-financial](https://github.com/chartjs/chartjs-chart-financial) plugin to support candlestick line types.  Since we're a .NET library, I've seen a lot of our users stick with Microsoft-compatible products too.  [ASP.NET Blazor](https://docs.microsoft.com/en-us/aspnet/core/blazor) is a good framework to consider since you can use our library directly in it.

Here's a few to explore:

- [Apache eCharts](https://echarts.apache.org/)
- [Charts.js](https://www.chartjs.org)
- [ScottPlot.NET](https://scottplot.net)
- [Plotly.NET](https://plotly.net/) or [Plotly.js](https://plotly.com/javascript/)
- [ApexCharts](https://apexcharts.com)
- [React Financial Charts](https://react-financial.github.io/react-financial-charts)
- [React Stockchart](http://rrag.github.io/react-stockcharts)
- [SyncFusion](https://www.syncfusion.com/)
- [Arction LightningCharts](https://www.arction.com)
- [TradingView Lightweight Charts](https://www.tradingview.com/lightweight-charts)
- [LiveCharts2](https://github.com/beto-rodriguez/LiveCharts2)
- [HighCharts](https://highcharts.com)
- [Google Charts](https://developers.google.com/chart)

What are some other ideas from the community?

**Comment by hieu-n** [comment]
I tried a few charting libraries for .Net. So far, Plotly.NET seems to be the best. It can even draws complex indicators like Ichimoku Cloud. It can output charts to a single HTML file, if you need something quick without starting a web server.

The downside is the document for .Net is quite a poor.

Another option I use is to output data to spreadsheets such as csv files or Google Sheets (via Google Sheet API), then draw charts using Excel or Google Sheets. Google Sheets can be connected to Looker Studio (aka Google Data Studio) and csv files can be used with PowerBi, if one needs a fancy dashboard.

**Comment by BizUnity** [comment]
I'm actually creating a stock library using ScottPlot, you can find it [here](https://github.com/BizUnity/StockPlot). I'm gona implement your indicators library on it, wonderfull job done here

**Reply by DaveSkender** [reply]
❤️ An easy to use charting plugin that accepts your own data feed would be nice.  Glad to hear you'll be using our library.  That'll save you years of work!

**Comment by grinay** [comment]
I’m adding this to the list because I found it before this topic and using the URL: http://klinecharts.com/

**Reply by DaveSkender** [reply]
Looks like the site defaults to Mandarin.
English version: https://klinecharts.com/en-US/
